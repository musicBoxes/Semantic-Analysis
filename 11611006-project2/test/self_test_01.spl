int while_for_test(){
	int i = 9;
	while (i){
		i = i + 1;
		while (i) break;
		break;
	}
	{
		int i = 0; // different scope
	}
	for ( ; ; ) break;
	for (int j = 0 ; ; ){
		int j = 0; // already def in for()
		{
			int j = 0; // different scope
		}
	}
	j = 0; // out of scope
	break; // break not in loop
	return 0;
}