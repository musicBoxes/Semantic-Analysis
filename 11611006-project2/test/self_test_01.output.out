Error type 3 at Line 13: Variable 'j' is redefined
Error type 1 at Line 18: Variable 'j' is not defined
Semantic Error at line 19: 'break' should be used in loop.
